packages/core/useElementVisibility/directive.ts

```diff
@@ -0,0 +1,24 @@
+import { watch } from 'vue-demi'
+import type { FunctionDirective } from 'vue-demi'
+import type { VisibilityScrollTargetOptions } from '.'
+import { useElementVisibility } from '.'
+
+type BindingValueFunction = (state: boolean) => void
+
+type BindingValueArray = [BindingValueFunction, VisibilityScrollTargetOptions]
+
+export const vElementVisibility: FunctionDirective<
+HTMLElement,
+BindingValueFunction | BindingValueArray
+> = (el, binding) => {
+  if (typeof binding.value === 'function') {
+    const handler = binding.value
+    const isVisible = useElementVisibility(el)
+    watch(isVisible, v => handler(v))
```

Comment:
Maybe we should triggered it immediately which that ensure the initial state is correctly.

Reply:
You're right!